src/live_restore/live_restore_fs.c

```diff
@@ -151,11 +151,7 @@ __live_restore_fs_create_stop_file(
     lr_fs = (WTI_LIVE_RESTORE_FS *)fs;
     path = path_marker = NULL;
 
-    /*
-     * FIXME-WT-14040: This is a big old race where we set this flag outside a lock and therefore
-     * have no idea if we will actually create a stop file after we clear the stop files.
-     */
-    if (lr_fs->finished)
+    if (__wti_live_restore_get_state(session, lr_fs) != WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION)
```

Comment:
Could we consider locking around this function so we don't need to check state twice. I know that may seem expensive but it's probably okay.

Reply:
Done in https://github.com/wiredtiger/wiredtiger/pull/11563/commits/a74f99d6276154eec22065d0ba3ba23fe238f2ff

Reply:
Add a `__wt_rwlock_islocked` assertion to that get state no_lock function if the expectation is that all callers call it locked.